cli/codegen.ts

```diff
@@ -0,0 +1,97 @@
+import { Project } from "ts-morph";
+import {
+  capitalize,
+  getDescriptors,
+  toDocs,
+} from "@fartlabs/ht/cli/codegen.ts";
+
+if (import.meta.main) {
+  const project = new Project();
+
+  // Create a new TSX component file for each descriptor.
+  const descriptors = getDescriptors();
+  for (const descriptor of descriptors) {
+    const sourceFile = project.createSourceFile(
+      `./${descriptor.tag}.tsx`,
+      "",
+      { overwrite: true, scriptKind: /* TSX */ 4 },
+    );
+
+    // Import the props interface.
+    sourceFile.addImportDeclaration(
+      descriptor.attrs.length === 0
+        ? {
+          isTypeOnly: true,
+          moduleSpecifier: "@fartlabs/ht/lib/global_attributes.ts",
+          namedImports: ["GlobalAttributes"],
+        }
+        : {
+          isTypeOnly: true,
+          moduleSpecifier: `@fartlabs/ht/${descriptor.tag}`,
+          namedImports: [descriptor.propsInterfaceName],
+        },
+    );
+
+    // Import the render function.
+    sourceFile.addImportDeclaration({
+      moduleSpecifier: `@fartlabs/ht/${descriptor.tag}`,
+      namedImports: [
+        { name: descriptor.functionName, alias: "render" },
+      ],
+    });
+
+    // Re-export the props interface.
+    sourceFile.addExportDeclaration({
+      isTypeOnly: true,
+      namedExports: [descriptor.propsInterfaceName],
+    });
+
+    // Create the component function.
+    const componentName = capitalize(descriptor.tag); // TODO: Consider using `.toUpperCase()` instead.
```

Comment:
### Considerations
- Lowercase component names are not syntactically recognized as JSX/TSX components.
- Pascal case component names are idiomatic JSX/TSX[^1].
- Uppercase component names (e.g. `BODY`) appear more like traditional HTML than pascal case component names (e.g. `Body`).

[^1]: https://eslint.style/rules/default/jsx-pascal-case

Reply:
I have decided to go with uppercase component names for this PR <https://github.com/FartLabs/htx/pull/1/commits/2264e620f242599e70dae9a32f19e2a2dd244125>.